src/query/query_opfunc.c

```diff
@@ -7237,28 +7243,11 @@ qdata_evaluate_aggregate_list (THREAD_ENTRY * thread_p, AGGREGATE_TYPE * agg_lis
 	      return ER_FAILED;
 	    }
 
-	  if ((DB_VALUE_DOMAIN_TYPE (&dbval) == DB_TYPE_STRING || DB_VALUE_DOMAIN_TYPE (&dbval) == DB_TYPE_VARNCHAR)
-	      && DB_GET_STRING_SIZE (&dbval) >= PRIM_MINIMUM_STRING_LENGTH_FOR_COMPRESSION)
+	  dbval_size = pr_data_writeval_disk_size (&dbval);
+	  if (dbval_size && (disk_repr_p = (char *) db_private_alloc (thread_p, dbval_size)))
```

Comment:
I'd like clearly rewrite it. 

```C
  if (dbval_size > 0 && (disk_repr_p = ...) != NULL)
```

Our coding convention does not recommend an assignment in `if`.
But, for this case, it will reduce a nesting depth.